engine/job.go

```diff
@@ -31,6 +34,7 @@ type jobImpl struct {
 	ttl       uint32
 	delay     uint32
 	tries     uint16
+	pool      string
```

Comment:
Why do we need this field? I think it will be better to support passing the pool name in `AddJob`

Reply:
Ok, will add another input argument in AddJob function.